internal/services/redis/redis_cache_resource.go

```diff
@@ -861,7 +861,12 @@ func expandRedisConfiguration(d *pluginsdk.ResourceData) (*redis.RedisCommonProp
 	// nolint : staticcheck
 	v, valExists = d.GetOkExists("redis_configuration.0.aof_backup_enabled")
 	if valExists {
-		output.AofBackupEnabled = utils.String(strconv.FormatBool(v.(bool)))
+		// aof_backup_enabled is available when SKU is Premium
+		if strings.EqualFold(skuName, string(redis.SkuNamePremium)) {
+			output.AofBackupEnabled = utils.String(strconv.FormatBool(v.(bool)))
+		} else if v.(bool) && !strings.EqualFold(skuName, string(redis.SkuNamePremium)) {
+			return nil, fmt.Errorf("The `aof_backup_enabled` property requires a `Premium` sku to be set")
+		}
```

Comment:
This can be simplified
```suggestion
		if v.(bool) && !strings.EqualFold(skuName, string(redis.SkuNamePremium)) {
			return nil, fmt.Errorf("the `aof_backup_enabled` property requires a `Premium` sku to be set")
		}
		output.AofBackupEnabled = pointer.To(strconv.FormatBool(v.(bool)))
```

Reply:
When the SKU is not "Premium", the default "false" will be sent to the service but this is not allowed by service

Reply:
yes, the SKU check is required.

Reply:
We need block the request when the SKU is `Premium` and `aof_backup_enabled` configured and also need tohandle the default values scenario as the bool type will default sets to `false`

Reply:
> We need block the request when the SKU is `Premium` and `aof_backup_enabled` configured

What is said here contradicts what's being done in this change.

I'm sorry but I'm not following anymore. Local testing isn't showing up any errors and the linked issue and PR have not given me any further insight.

Can you please add an acceptance test that demonstrates the problem this is trying to fix?

Reply:
sure，  I will add the test to cover this scenario

Reply:
@stephybun  Code updated and add a test case.